lib/router/mbaas-auth.js

```diff
@@ -0,0 +1,36 @@
+const q = require('q');
+
+/**
+ * Auth sends a request on 'wfm:user:auth' topic, raincatcher-demo-auth is subscriber
+ * of this topic and verifies passed in credentials.
+ *
+ * @param {Object} mediator object used to subscribe and publish read and auth topics
+ * @param {String} username Unique identifier of the user.
+ * @param {String} password passed in password to be verified against user.
+ *
+ */
+exports.auth = function(mediator, username, password) {
+
+  /**
+   * Returns promise with user profile for authenticated user, or an error
+   * if user doesn't exist or is not authenticated.
+   *
+   * @param {boolean} authenticated
+   * @returns {*|promise} containing user profile for authenticated user.
+   */
+  function checkAuthentication(authenticated) {
+    if (authenticated) {
+      console.log('authenticated');
+      return mediator.request('wfm:user:username:read', username);
+    }
+    var checkAuthDeferred = q.defer();
+    console.log('Invalid Credentials');
+    checkAuthDeferred.reject(new Error('Invalid Credentials'));
+    return checkAuthDeferred.promise;
+  }
+
+  console.log('Checking credentials for user');
+  // this will return true/false result of verification or User not found error
+  return mediator.request('wfm:user:auth', {username: username, password: password}, {uid: username})
```

Comment:
The `:auth` topic actually takes in a username as opposed to a unique id, since it deals with user:password authentication, maybe we should be looking into renaming the topic to something more specific instead?